### PR TITLE
viruses with the inorganic biology symptom can now infect androids

### PR DESCRIPTION
**Pull request by ATH1909**
## About The Pull Request

The inorganic biology symptom now adds both MOB_MINERAL and MOB_ROBOTIC to a virus's list of infectable biotypes instead of only MOB_MINERAL. Currently, androids are the only species this change affects.

## Why It's Good For The Game

Robots are inorganic too!

EDIT: To be clear: No, inorganic biology viruses won't be able to infect borgs or AI cores, that's not how this works.

## Changelog

:cl: ATHATH
balance: The inorganic biology symptom now adds both MOB_MINERAL and MOB_ROBOTIC to a virus's list of infectable biotypes instead of only MOB_MINERAL. Currently, androids are the only species this change affects.
/:cl:

### CODE REVIEW

File: code/datums/diseases/advance/symptoms/species.dm [hunk 1]
```diff
@@ -44,7 +44,9 @@
 
 /datum/symptom/inorganic_adaptation/OnAdd(datum/disease/advance/A)
 	A.infectable_biotypes |= MOB_MINERAL //Mineral covers plasmamen and golems.
+	A.infectable_biotypes |= MOB_ROBOTIC //Robotic covers androids.
```
Comment (Jacquerel) [inline code/datums/diseases/advance/symptoms/species.dm:47]:
```suggestion
	A.infectable_biotypes |= MOB_MINERAL | MOB_ROBOTIC // Plasmamen, Golems, and Androids.
```

File: code/datums/diseases/advance/symptoms/species.dm [hunk 2]
```diff
@@ -44,7 +44,9 @@
...
 
 /datum/symptom/inorganic_adaptation/OnRemove(datum/disease/advance/A)
 	A.infectable_biotypes &= ~MOB_MINERAL
+	A.infectable_biotypes &= ~MOB_ROBOTIC
```
Comment (Jacquerel) [inline code/datums/diseases/advance/symptoms/species.dm:51]:
```suggestion
	A.infectable_biotypes &= ~(MOB_MINERAL | MOB_ROBOTIC)
```

**Review by Jacquerel — CHANGES_REQUESTED** [review]

**Comment by ATH1909** [comment]
> You haven't really explained why this is good for the game. "It makes sense" is sort of insufficient in of itself, although given that androids are virtually nonexistent it's not going to be hard to justify.

robots are inorganic, and the description of the symptom doesn't explicitly blacklist them.

as you said, androids are rare, so this is basically just a consistency/edge case thing, hence why I've labeled this as a "fix" instead of "balance".

**Comment by softcerv** [comment]
Could this lead to symptoms like nocturnal regeneration working on borgs?

**Comment by ATH1909** [comment]
> Could this lead to symptoms like nocturnal regeneration working on borgs?

..............................................................................

image

I thought I was going too far by explicitly spelling that out, but apparently people are still clueless about how viro works

**Comment by softcerv** [comment]
> > Could this lead to symptoms like nocturnal regeneration working on borgs?
> 
> ..............................................................................
> 
> image
> 
> I thought I was going too far by explicitly spelling that out, but apparently people are still clueless about how viro works

Right, that's my bad. I'll ask another question then. How much testing have you done to ensure that all of the symptoms work on androids without creating runtime errors or causing unintended behaviors?

**Comment by ATH1909** [comment]
> Right, that's my bad. I'll ask another question then. How much testing have you done to ensure that all of the symptoms work on androids without creating runtime errors or causing unintended behaviors?

the necrotic metabolism symptom works on skeletons, who also don't process most reagents

I... don't really get what you're implying here?

**Pull request merged**